### PR TITLE
Make git_status picker async Fixes #2714

### LINKED ISSUE #2714: configurable timeout for `:Telescope git_status`
**Is your feature request related to a problem? Please describe.**
I work in a very large git repo, in which git operations often take a long time. This often causes `:Telescope git_status` to time out.

**Describe the solution you'd like**
A configurable timeout setting for the `git_status` telescope builtin.

**Describe alternatives you've considered**
I couldn't think of any other alternatives.

**Additional context**

### PR DESCRIPTION
**Pull request by lrfurtado**
# Description

Convert the git status from being a synchronous job that had timeout issues on low-performance networks or really large repos. To sue async_oneshot_job
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Requested the reporter to perform tests on his machine. I also tested locally with a git wrapper that introduced fake latency,

**Configuration**:
* Neovim version (nvim --version):
* Operating system and version:

NVIM v0.9.2
Build type: Release
LuaJIT 2.1.0-beta3

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/0.9.2/share/nvim"

Run :checkhealth for more info

# Checklist:

- [x] My code follows the style guidelines of this project (stylua)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (lua annotations)

### CODE REVIEW

File: lua/telescope/builtin/__git.lua [hunk 1]
```diff
@@ -397,9 +383,21 @@ git.status = function(opts)
   pickers
     .new(opts, {
       prompt_title = "Git Status",
+      entry_maker = opts.entry_maker,
```
Comment (Conni2461) [inline lua/telescope/builtin/__git.lua:386]: this looks wrong, entry_maker are defined on finder not on pickers

File: lua/telescope/builtin/__git.lua [hunk 2]
```diff
@@ -397,9 +383,21 @@ git.status = function(opts)
...
+        function(self)
+          local lines = self.manager:num_results()
+          if lines == 0 then
+            utils.notify("builtin.git_status", {
+              msg = "No changes found",
+              level = "ERROR",
+            })
+          end
+        end,
+      },
```
Comment (Conni2461) [inline lua/telescope/builtin/__git.lua:400]: this part here is interesting, i've never thought about this, how we should handle things if we dont have any results with a oneshot finder.

do you think it makes sense to then also close the picker, if we havent found anything?

Comment (lrfurtado) [reply lua/telescope/builtin/__git.lua]: @Conni2461 thanks for the feedback I have addressed your comments and updated the PR.

**Comment by Conni2461** [comment]
lgtm thanks. but i've noticed that the picker not flickers when using `<Tab>`, but i dont think this is a big deal

**Comment by lrfurtado** [comment]
@Conni2461 the user has reported this PR addreses the problem.

**Pull request merged**